adapter/statsd/statsd.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"text/template"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/golang/protobuf/ptypes/duration"
+	"istio.io/mixer/adapter/statsd/config"
+	"istio.io/mixer/pkg/adapter"
+)
+
+const (
+	defaultFlushBytes = 512
+)
+
+type (
+	builder struct {
+		adapter.DefaultBuilder
+	}
+
+	aspect struct {
+		rate      float32
+		client    statsd.Statter
+		templates map[string]*template.Template // metric name -> template
+	}
+)
+
+var (
+	name        = "statsd"
+	desc        = "Pushes statsd metrics"
+	defaultConf = &config.Params{
```

Comment:
If there's a non-zero chance someone would just prop up a default statsd service and a default mixer to use it and things just work, then why not make that the default address?